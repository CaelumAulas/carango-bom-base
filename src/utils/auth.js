import AuthService from "../services/Auth/AuthService";

const estaLogado = async () => {
  const token = localStorage.getItem("token");

  const resposta = await AuthService.validar(token);

  if(resposta) {
    return true;
  }

  localStorage.clear();
  return false;
}

export { estaLogado }