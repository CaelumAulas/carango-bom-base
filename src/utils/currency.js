export function formatCurrency(currency) {
  const currencyNumbersStr = currency.replace(/[.,]/g, '')

  let currencyNum = Number(currencyNumbersStr)
  let decimalPart = `,${currencyNum % 100}`
  let currencyValue = Math.trunc(currencyNum / 100)
  
  let formattedCurrency = ''
  
  while(currencyValue % 1000 !== currencyValue) {
    formattedCurrency = `.${currencyValue % 1000}${formattedCurrency}`
    currencyValue = Math.trunc(currencyValue / 1000)
  }

  return `${currencyValue}${formattedCurrency}${decimalPart}`
}

export function getOnlyNumbers(currency) {
  return currency.replace(/\D+/g, "")
}