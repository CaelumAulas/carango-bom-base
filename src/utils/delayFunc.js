export function delayFunc(func, milliseconds = 1000) {
  return new Promise(resolve => {
    setTimeout(() => {
      func()
      resolve()
    }, milliseconds)
  })
}