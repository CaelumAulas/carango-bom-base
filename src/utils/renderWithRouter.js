import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

export function renderWithRouter(Component) {
  return (
    render(<MemoryRouter>{Component}</MemoryRouter>)
  )
}