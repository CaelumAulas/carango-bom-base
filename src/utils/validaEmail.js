export function validaEmail(email) {
  const regex = /^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$/g
  return !!email.match(regex)
}