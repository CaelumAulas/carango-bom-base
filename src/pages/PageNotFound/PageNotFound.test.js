import { screen } from "@testing-library/react"
import { PageNotFound } from "."
import { renderWithRouter } from "../../utils/renderWithRouter"


describe('<PageNotFound />', () => {
  it('should render correctly', () => {
    const { container } = renderWithRouter(<PageNotFound />)

    expect(screen.getByRole('heading', { name: /página não encontrada/i})).toBeInTheDocument()
    expect(screen.getByRole('link', { name: /voltar para o início/i})).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})