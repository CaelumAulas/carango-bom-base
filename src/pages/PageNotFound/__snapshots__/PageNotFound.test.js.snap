// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNotFound /> should render correctly 1`] = `
<div>
  <section
    class="makeStyles-root-1"
  >
    <h1>
      Página não encontrada :(
    </h1>
    <a
      class="makeStyles-link-2"
      href="/"
    >
      Voltar para o início
    </a>
  </section>
</div>
`;
